docs/design/content-type-icon.md

```diff
@@ -0,0 +1,207 @@
+
+* `apps/package_network.png`
+* `apps/package_settings.png`
+* `apps/package.png`
+* `apps/kuser.png`
+* `apps/acroread.png`
+* `apps/personal.png`
+* `apps/locale.png`
+* `actions/identity.png`
+* `actions/view_tree.png`
+* `filesystems/folder_image.png`
+* `filesystems/folder_video.png`
+* `filesystems/folder_man.png`
+* `filesystems/folder_txt.png`
+* `filesystems/folder.png`
+* `mimetypes/exec_wine.png`
+* `mimetypes/tar.png`
+* `mimetypes/sound.png`
+* `mimetypes/pdf.png`
+* `mimetypes/access.png`
+* `mimetypes/real.png`
+* `mimetypes/quicktime.png`
+* `mimetypes/visio.png`
+* `mimetypes/binary.png`
+* `mimetypes/txt.png`
+* `mimetypes/rpm.png`
+* `mimetypes/word.png`
+* `mimetypes/tex.png`
+* `mimetypes/log.png`
+* `mimetypes/txt2.png`
+* `mimetypes/excel.png`
+* `mimetypes/readme.png`
+* `mimetypes/video.png`
+* `mimetypes/dvi.png`
+* `mimetypes/document.png`
+* `mimetypes/real_doc.png`
+* `mimetypes/powerpoint.png`
+* `mimetypes/html.png`
+* `mimetypes/ascii.png`
+* `mimetypes/empty.png`
+* `mimetypes/tgz.png`
+* `mimetypes/wordprocessing.png`
+* `mimetypes/publisher.png`
+* `mimetypes/image.png`
+
+## Font icons + CSS
+
+The icons in the PlatformUI interface are provided by an icon font. For Content
+Type, the idea is to expand that concept so that while generating the interface,
+we end up with a code similar to:
+
+```html
+<h1 class="ez-contenttype-icon ez-contenttype-icon-folder">Folder Name</h1>
+```
+
+With such classes, the `h1` is specified to display a Content Type icon. The
+class `ez-contenttype-icon` makes sure the element is style for that and get the
+default Content Type icon. The second class is specific to the Content Type
+based on its identifier and if it's defined in one of the CSS file, the element
+will get the custom Content Type icon defined there.
+
+## Extensibility
+
+The extensibility is tackled differently depending on the use case but it relies
+on the ability to embed a custom CSS file in PlatformUI with `css.yml`.
+
+Also to even prevent the need to configure/extend the system, we should provide
+several pre-configured icons for very common Content Types like:
+
+* `product`
+* `author`
+* `category`
+* `gallery` / `portfolio`
+* `blog_post` / `blogpost` / `post`
+* `blog` / `weblog`
+* `news`
+* `pdf`
+* `document`
+* `photo`
+* `comment`
+* `wiki`
+* `wiki_page` / `wikipage`
+
+
+### Pick an icon for a custom Content Type in existing icons
+
+In that case, you need to pick the icon code. For that, the icomoon application
+can be used until the UI guideline is up to date and references the available
+icons. To ease that process and the readability of the code, we'll use
+ligatures in the font icon so that the CSS code for a custom Content Type could
+look like:
+
+```css
+/* in a custom CSS file included with `css.yml` */
+
+.ez-contenttype-icon-mycontenttypeidentifier:before {
+    content: "product"; /* because this icon matches the usage of such content
```

Comment:
it's a *ligature*, basically the identifier of an icon. We could also write something like:
```css
.ez-contenttype-icon-mycontenttypeidentifier:before {
    content: "\E642";
}
```